### ISSUE TITLE
Serbian preset info from iD is not being pulled in

### ISSUE THREAD

**Issue by matkoniecz** [streetcomplete/StreetComplete#3897]
That is because `bcp47ExportLanguages` has it defined as `sr-cyrl` while iD presets use `sr` and is skipped in https://github.com/streetcomplete/StreetComplete/blob/ed43396b5820f69bb277bf68fef27073fd9d25ee/buildSrc/src/main/java/UpdatePresetsTask.kt#L30

See https://www.transifex.com/openstreetmap/id-editor/language/sr/ - presets are fully translated

missing from https://github.com/streetcomplete/StreetComplete/tree/master/app/src/main/assets/osmfeatures/default

Maybe this and Norwegian should rather use modified https://github.com/streetcomplete/StreetComplete/blob/6aca0dc9476034466d04949ffe870075932edb96/buildSrc/src/main/java/Utils.kt

**Comment by HolgerJeromin** [comment]
IMO we should split both language strings at "-" and try to match the first part if the exact string is not found.
That way this would be more generic

**Comment by matkoniecz** [comment]
First part, needed to obtain iD preset data is in https://github.com/matkoniecz/Zazolc/commit/01d95377743fd299a8ad6420c32d2e0836e82e3a

I am tempted to add just one more kludge and move `sr-Cyrl` to `sr` in https://github.com/streetcomplete/StreetComplete/blob/ed43396b5820f69bb277bf68fef27073fd9d25ee/buildSrc/src/main/java/UpdatePresetsTask.kt#L12

> IMO we should split both language strings at "-" and try to match the first part if the exact string is not found.

This could be accomplished by simply using already existing `javaLanguageTagToAndroidResCodes` in UpdatePresetsTask  

```
val javaLanguage = bcp47LanguageTagToJavaLanguageTag(language)
File("$targetDir/$javaLanguage.json").writeText(presetsLocalization)
```

could be using `javaLanguageTagToAndroidResCodes` as extra tranformation step.


But I am not sure about potential side-effects.

@westnordost - why in `javaLanguageTagToAndroidResCodes` ( https://github.com/streetcomplete/StreetComplete/blob/6aca0dc9476034466d04949ffe870075932edb96/buildSrc/src/main/java/Utils.kt ) there is `if (languageTag == "zh-CN") return listOf("zh")`?

From what I see `zh-CN` is actually using `zh-CN` files, at least for iD presets.

**Comment by westnordost** [comment]
I also noticed that osmfeatures had support for scripts (e.g. "bg-Cyrl" missing). Fixed that and v4.1.
